### PR TITLE
Release 5.6.0

### PR DESCRIPTION
**Pull request by weeeBox**
#### Improvements

* Google Play In-App Review support.

### CODE REVIEW

**Review by melj32app — APPROVED** [review]
Looks good to me.

File: apptentive/src/main/java/com/apptentive/android/sdk/external/GooglePlayReviewManager.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    private void launchReviewFlow(@NonNull final Activity activity, ReviewInfo reviewInfo, final InAppReviewListener callback) {
+        final long startTime = System.currentTimeMillis();
+
+        Task<Void> task = reviewManager.launchReviewFlow(activity, reviewInfo);
+        task.addOnSuccessListener(new OnSuccessListener<Void>() {
+            @Override
+            public void onSuccess(Void result) {
+                long elapsedTime = System.currentTimeMillis() - startTime;
+                if (elapsedTime < 1000L) {
+                    notifyFailure(callback, null, "In-app review flow completed too fast (%d ms) and we have good reasons to believe it just failed silently.", elapsedTime); // FIXME: better error message
```
Comment (fkose696) [inline apptentive/src/main/java/com/apptentive/android/sdk/external/GooglePlayReviewManager.java:79]: I get this every time I engage an event. I think it's a breaking change as we're losing the rating dialog all together in this scenario.

Comment (weeeBox) [reply apptentive/src/main/java/com/apptentive/android/sdk/external/GooglePlayReviewManager.java:79]: fkose696, `elapsedTime` is the **total** time measured from the moment we send a request til the moment user dismisses the dialog. Usually, it's something around 10-60 seconds. We picked 1 second for a couple of reasons:
- dialog UI show/hide animation takes 400 ms, there's also a lag caused by launching new activity ~200ms, then user has to press 'back' button on click on 'Not now' button. It's literally impossible to do this under a second.
- There's **no reliable way to tell if the rating prompt was shown or not** (besides using private API which tends to be a poor idea) but we found that timing approach works reasonable well.

Make sure you have configured your environment correctly:
- The user account is part of the Internal Test Track.
- The user account is the primary account and it's selected in the Play Store.
- The user account has downloaded the app from the Play Store (the app is listed in the user's Google Play library).
- The user account **does not currently have a review** for the app.

We also experienced random failures even when all the above is true - in this case you can try the following:
- Clear Google Play application data
- Re-install Google Play application data
- Sign-out and sign-in in your account.
- Perform device factory reset.

If none of above works - ship the update and see if it works in production.

For more information: https://developer.android.com/guide/playcore/in-app-review/test

Hope, this helps!

Comment (fkose696) [reply apptentive/src/main/java/com/apptentive/android/sdk/external/GooglePlayReviewManager.java:79]: I'll have to set up the internal test track and re-test. Thanks for the prompt response!

Comment (weeeBox) [reply apptentive/src/main/java/com/apptentive/android/sdk/external/GooglePlayReviewManager.java:79]: No problem!

**Pull request merged**